cdap-ui/app/features/dashboard/model/dashboards-model.js

```diff
@@ -58,23 +61,16 @@ function (Widget, MyDataSource, mySettings, $q) {
    */
   Dashboard.prototype.removeWidget = function (widget) {
     var that = this;
-    angular.forEach(that.columns, function (c, i) {
-      that.columns[i] = c.filter(function (p) {
-        return widget !== p;
-      });
+    this.columns = this.columns.filter(function(p) {
+      return widget !== p;
     });
     this.checkForEmptyDashboard();
     this.persist();
   };
 
   Dashboard.prototype.checkForEmptyDashboard = function() {
-    var isEmpty = true;
-    angular.forEach(this.columns, function(column) {
-      if (column.length > 0) {
-        isEmpty = false;
-      }
-    });
-    this.isEmpty = isEmpty;
+    this.isEmpty = this.columns.length === 0;
+    return this.columns.length === 0;
```

Comment:
Fixed this in this PR (https://github.com/caskdata/cdap/pull/2707) . This PR was supposed to be adding only angular-gridster. PR #2707  was supposed to be for re-organization  + cleanup.